### PR TITLE
feat(websites): soap2day

### PR DESCRIPTION
**Pull request by fuwn**

### CODE REVIEW

**Comment by EncryptedDev** [comment]
Upload proof of it working?

**Comment by fuwn** [comment]
> Upload proof of it working?

https://github.com/PreMiD/Presences/pull/3435#issuecomment-850864643, or would you prefer a video/ gif of the presence in action?

**Comment by ririxidev** [comment]
@fuwn that's okay.

File: websites/S/SOAP2DAY/presence.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    return "tvshow";
+  } else if (window.location.href.indexOf("Mczo") != -1) {
+    return "movie";
+  } else if (window.location.href.indexOf("Sczo") != -1) {
+    return "sport";
+  } else if (window.location.href.indexOf("faq") != -1) {
+    return "faq";
+  } else if (window.location.href.indexOf("Eczo") != -1) {
+    return "tv";
+  } else { return "home" }
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:21]: Use `!==` and remove unnecessary curly brackets.

File: websites/S/SOAP2DAY/presence.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  return document.getElementsByClassName(text)[0].textContent.trim();
+},
+epochWithOffset = (t: number): number => {
+  const now = new Date();
+  now.setTime(now.getTime() + t);
+  return now.setTime(now.getTime());
+},
+getStatus = (): string => {
+  const element = document.getElementById("t3").textContent.trim();
+  if (element === "") { return "Loading" } else { return element }
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:33]:
```suggestion
  if (element === "") return "Loading";
  else return element;
```

File: websites/S/SOAP2DAY/presence.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    details: constructAction[getAction()]
+  };
+  // If the user is watching something, get the title and set duration.
+  if (["movie", "tv", "sport"].includes(getAction())) {
+    // If paused, reset update remaining.
+    if (getStatus() == "Pause") { flag = false }
+
+    if (!flag) {
+      watchStamp = epochWithOffset(document.getElementsByTagName("video")[0].duration);
+      if (!isNaN(watchStamp)) { flag = true }
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:62]:
```suggestion
    if (getStatus() === "Pause") flag = false;

    if (!flag) {
      watchStamp = epochWithOffset(document.getElementsByTagName("video")[0].duration);
      if (!isNaN(watchStamp)) flag = true;
```

File: websites/S/SOAP2DAY/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/dist/metadata.json:2]:
```suggestion
  "$schema": "https://schemas.premid.app/metadata/1.4",
```

File: websites/S/SOAP2DAY/presence.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      ...presenceData
+    };
+  }
+
+  if (presenceData.details == null) {
+    presence.setTrayTitle();
+    presence.setActivity();
+  } else {
+    presence.setActivity(presenceData);
+  }
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:82]: Ok so about this, the details is null-able? According to your code it shouldn't be null-able

**Review by SlowLife1661 — CHANGES_REQUESTED** [review]
It would be good if you have smallImage for example when the video is pauses or plays

File: websites/S/SOAP2DAY/presence.ts (hunk 5 above)
Comment (fuwn) [reply websites/S/SOAP2DAY/presence.ts:75]: I literally just nabbed this from the PreMiD presence development documentation, no clue either.

Comment (SlowLife1661) [reply websites/S/SOAP2DAY/presence.ts:75]: Well.., basically are you supporting all sub-pages

Comment (fuwn) [reply websites/S/SOAP2DAY/presence.ts:75]: How do you mean?

Comment (fuwn) [reply websites/S/SOAP2DAY/presence.ts:75]: If by that you mean all categories of SOAP2DAY, yes, I am.

Comment (SlowLife1661) [reply websites/S/SOAP2DAY/presence.ts:75]:
```suggestion
presence.setActivity(presenceData);
```

**Comment by SlowLife1661** [comment]
Also proof of working? I haven't seen your Presence so far

**Comment by fuwn** [comment]
> Also proof of working? I haven't seen your Presence so far

[](https://strelizia.cc/eZ4tC8ax3oWFW8xa0y2w4x51x6CR5rW2.mp4)

File: websites/S/SOAP2DAY/presence.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  };
+  // If the user is watching something, get the title and set duration.
+  if (["movie", "tv", "sport"].includes(getAction())) {
+    // If paused, reset update remaining.
+    if (getStatus() === "Pause") pauseFlag = false;
+
+    if (pauseFlag) {
+      watchStamp = (document.getElementsByTagName("video")[0].duration -
+        document.getElementsByTagName("video")[0].currentTime)
+        + Math.floor(Date.now() / 1000);
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:50]:
```suggestion
      watchStamp = presence.getTimestampsfromMedia(document.getElementsByTagName("video")[0])[1]
```

File: websites/S/SOAP2DAY/presence.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    largeImageKey: "icon",
+    details: constructAction[getAction()]
+  };
+  // If the user is watching something, get the title and set duration.
+  if (["movie", "tv", "sport"].includes(getAction())) {
+    // If paused, reset update remaining.
+    if (getStatus() === "Pause") pauseFlag = false;
+
+    if (pauseFlag) {
+      watchStamp = presence.getTimestampsfromMedia(document.getElementsByTagName("video")[0])[1]
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:48]: My bad
```suggestion
      watchStamp = presence.getTimestampsfromMedia(document.getElementsByTagName("video")[0])[1];
```

File: websites/S/SOAP2DAY/presence.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        ...presenceData
+      };
+    }
+    pauseFlag = true;
+  } else { // If the user is not watching something, return how long they have been browsing.
+    presenceData = {
+      startTimestamp: Math.floor(Date.now() / 1000),
+      ...presenceData
+    };
+  }
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:69]: Ok last thing, remove curly brackets on this else statement

**Review by SlowLife1661 — CHANGES_REQUESTED** [review]

File: websites/S/SOAP2DAY/presence.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+      ...presenceData
+    };
+    
+    pauseFlag = true;
+  } else { // If the user is not watching something, return how long they have been browsing.
+    presenceData = {
+      startTimestamp: Math.floor(Date.now() / 1000),
+      ...presenceData
+    };
+  }
```
Comment (SlowLife1661) [inline websites/S/SOAP2DAY/presence.ts:68]: This
```suggestion
  } else // If the user is not watching something, return how long they have been browsing.
    presenceData = {
      startTimestamp: Math.floor(Date.now() / 1000),
      ...presenceData
    };
```

**Review by SlowLife1661 — DISMISSED** [review]
Looks good to me 🙂

**Review by SlowLife1661 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]
LGTM

**Pull request merged**